lib/tcp-server.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const net = require('net');
+const server = net.createServer;
+const sockets = [];
+
+server((socket)=>{
+  socket.ID = 'User' + Math.floor(Math.random() * 1000);
```

Comment:
This works as far as attaching an unique value to the socket instance.  There's always a question as to whether the socket already has a property called ID that's you'd be overwriting.  You could test for this (if !socket.ID), check docs,  or you could pick a property name that is more specific - like MYAPP_ID.  You could also just use a local var with the index of the array  like "user " + index.  You know index as length -1 after the push.